sandbox.go

```diff
@@ -353,7 +395,7 @@ func (sb *sandbox) clearNetworkResources(ep *endpoint) error {
 	delete(sb.epPriority, ep.ID())
 	sb.Unlock()
 
-	if highEpBefore != highEpAfter {
+	if sb.osSbox != nil && highEpBefore != highEpAfter {
```

Comment:
Can we do the `sb.osSbox != nil` check in `UpdateGateway` with proper locking for `sb` before we do anything in `UpdateGateway`. That way we protect it where we need it and we don't need to do this again and again in every caller site.